src/python/pants/backend/python/macros/poetry_requirements.py

```diff
@@ -330,10 +330,16 @@ def parse_pyproject_toml(pyproject_toml: PyProjectToml) -> set[Requirement]:
     # See: https://python-poetry.org/docs/pyproject/#dependencies-and-dev-dependencies
     dependencies.pop("python", None)
 
+    groups = poetry_vals.get("group", {})
+    group_deps = {}
+
+    for group in groups.values():
+        group_deps.update(**group.get("dependencies", {}))
```

Comment:
No need to unpack w/ `**`. `update()` takes a dictionary as an arg.

```suggestion
    group_deps = {}
    for group in groups.values():
        group_deps.update(group.get("dependencies", {}))
```